recipes/msix/all/test_package/example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <AppxPackaging.hpp>
+
+LPVOID STDMETHODCALLTYPE MyAllocate(SIZE_T cb) {
+	return std::malloc(cb);
+}
+
+void STDMETHODCALLTYPE MyFree(LPVOID pv) {
+	std::free(pv);
+}
+
+int main() {
+	IAppxFactory* appxFactory;
+
+	HRESULT creationResult = CoCreateAppxFactoryWithHeap(
+		MyAllocate,
+		MyFree,
+		MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_FULL,
+		&appxFactory
+	);
+
+	if (FAILED(creationResult)) {
```

Comment:
Does this work on Linux? Isn't `FAILED` something Windows SDK specific?

Reply:
In general, `FAILED` is [a macro in `winerror.h`](https://docs.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-failed) but `msix` has the special public header [`MSIXWindows.hpp`](https://github.com/microsoft/msix-packaging/blob/v1.7/src/inc/MSIXWindows.hpp) that works like a shim.